query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -96,6 +95,18 @@ private String functionToStringForAvgRate(
         columnName, (double) aggregateIntervalInSeconds / rateIntervalInSeconds);
   }
 
+  private Optional<Expression> getColumnNameForAvgRate(Function function) {
```

Comment:
Why are we returning an Optional here if the caller is forcefully doing get()?
Optional here makes sense if the caller needs to deal with the value being present or not.

Reply:
Agreed. Also, this isn't returning a column name, it's returning an expression. If I sent my args out of order, this would still work which is not what we want. I would just change the old line 84 to default if empty, and leave the rest as is.

Reply:
resolved